### PR TITLE
PHOENIX-6859 Update phoenix5-spark3 README with PySpark code references

### PR DESCRIPTION
**Pull request by Abhey**
This PR introduces the changes for adding the PySpark code references in the phoenix5-spark3 connector README.

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -1 :x: |  markdownlint  |   0m  1s |  The patch generated 30 new + 50 unchanged - 4 fixed = 80 total (was 54)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |   7m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/92 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux 8a914290e3a8 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / 4da44c7 |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/1/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark3 U: phoenix5-spark3 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/1/console |
| versions | git=2.7.4 maven=3.3.9 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -1 :x: |  markdownlint  |   0m  2s |  The patch generated 33 new + 46 unchanged - 8 fixed = 79 total (was 54)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |   2m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/92 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux fdceb27a730b 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / 4da44c7 |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/2/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/2/artifact/yetus-general-check/output/whitespace-eol.txt |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark3 U: phoenix5-spark3 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/2/console |
| versions | git=2.7.4 maven=3.3.9 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  4s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -1 :x: |  markdownlint  |   0m  2s |  The patch generated 10 new + 28 unchanged - 26 fixed = 38 total (was 54)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   |   2m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/92 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux fb3451f06b7c 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / 4da44c7 |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/3/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| whitespace | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/3/artifact/yetus-general-check/output/whitespace-eol.txt |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark3 U: phoenix5-spark3 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/3/console |
| versions | git=2.7.4 maven=3.3.9 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  6s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| -1 :x: |  markdownlint  |   0m  1s |  The patch generated 3 new + 19 unchanged - 35 fixed = 22 total (was 54)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 31s |  The patch does not generate ASF License warnings.  |
|  |   |   2m  4s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/92 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux 9e5b71556a5d 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / 4da44c7 |
| markdownlint | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/4/artifact/yetus-general-check/output/diff-patch-markdownlint.txt |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark3 U: phoenix5-spark3 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/4/console |
| versions | git=2.7.4 maven=3.3.9 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  0s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  markdownlint  |   0m  2s |  The patch generated 0 new + 15 unchanged - 39 fixed = 15 total (was 54)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 30s |  The patch does not generate ASF License warnings.  |
|  |   |   1m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/92 |
| Optional Tests | dupname asflicense markdownlint |
| uname | Linux 822a16ddacb3 4.15.0-200-generic #211-Ubuntu SMP Thu Nov 24 18:16:04 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / 4da44c7 |
| Max. process+thread count | 47 (vs. ulimit of 30000) |
| modules | C: phoenix5-spark3 U: phoenix5-spark3 |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-92/5/console |
| versions | git=2.7.4 maven=3.3.9 markdownlint=0.22.0 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Abhey** [comment]
@virajjasani @tkhurana Please review the PR.

**Comment by virajjasani** [comment]
@stoty you might also be interested in this

**Comment by virajjasani** [comment]
Thanks for the PR @Abhey

**Comment by stoty** [comment]
Would it be possible to add the relevant changes to the Spark2 README ?

File: phoenix5-spark3/README.md [hunk 1]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-phoenix-spark extends Phoenix's MapReduce support to allow Spark to load Phoenix tables as DataFrames,
-and enables persisting DataFrames back to Phoenix.
+# Phoenix5-Spark3 Connector
 
-## Configuring Spark to use the connector
+The phoenix5-spark3 plugin extends Phoenix's MapReduce support to allow Spark
```
Comment (stoty) [inline phoenix5-spark3/README.md:20]: I realize the we use "Plugin" on the website, but we should standardize on "Connector"

File: phoenix5-spark3/README.md [hunk 2]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
...
+## Why not JDBC
+
+Although Spark supports connecting directly to JDBC databases,
+ It’s only able to parallelize queries by partioning on a numeric column.
+ It also requires a known lower bound,
+ upper bound and partition count in order to create split queries.
+
+In contrast, the phoenix-spark integration is able to leverage the underlying
+ splits provided by Phoenix in order to retrieve and save data across multiple
+ workers. All that’s required is a database URL and a table name.
```
Comment (stoty) [inline phoenix5-spark3/README.md:38]: use "select statement" instead of "table name" ?

File: phoenix5-spark3/README.md [hunk 3]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
...
+ Optional SELECT columns can be given,
+ as well as pushdown predicates for efficient filtering.
```
Comment (stoty) [inline phoenix5-spark3/README.md:40]: This sounds like you need tospecify the pushdown predicates.
Can you rephrase so that it's apparent that pushdown is automatic ?

File: phoenix5-spark3/README.md [hunk 4]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
...
+
+The choice of which method to use to access
+ Phoenix comes down to each specific use case.
```
Comment (stoty) [inline phoenix5-spark3/README.md:43]: nit:
This is super important, and we should have much more on this (though not necessarily in this ticket)

File: phoenix5-spark3/README.md [hunk 5]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
...
+
+## Setup
+
+To setup connector add `phoenix5-spark3-shaded` JAR as
```
Comment (stoty) [inline phoenix5-spark3/README.md:47]: In most cases, you don't want to add the connector to the maven/compile classpath, it tends to cause conflicts when upgrading.

We should move this to emd of the section, and add the caveat that this is only needed for the deprecated usages.

File: phoenix5-spark3/README.md [hunk 6]
```diff
@@ -15,18 +15,59 @@ See the License for the specific language governing permissions and
...
+In contrast, the phoenix-spark integration is able to leverage the underlying
+ splits provided by Phoenix in order to retrieve and save data across multiple
+ workers. All that’s required is a database URL and a table name.
+ Optional SELECT columns can be given,
+ as well as pushdown predicates for efficient filtering.
+
+The choice of which method to use to access
+ Phoenix comes down to each specific use case.
+
+## Setup
```
Comment (stoty) [inline phoenix5-spark3/README.md:45]: Nit: this assumes that Phoenix and HBase/Spark are are both present and configured on the same nodes.
Maybe worth mentioning it ?

File: phoenix5-spark3/README.md [hunk 7]
```diff
@@ -39,7 +80,9 @@ UPSERT INTO TABLE1 (ID, COL1) VALUES (2, 'test_row_2');
 ```
 
 ### Load as a DataFrame using the DataSourceV2 API
+
 Scala example:
+
```
Comment (stoty) [inline phoenix5-spark3/README.md:85]: I know you didn't touch that part, but do we still need the SparkContext import ?

Comment (stoty) [inline phoenix5-spark3/README.md:85]: Maybe add comments to make  it obvious that you need to use a real ZK quorum , like 
`//replace "phoenix-server:2181" with the real ZK quorum`

**Review by stoty — CHANGES_REQUESTED** [review]

**Comment by Abhey** [comment]
Thanks, @stoty for reviewing the PR. I will take a look at your review comments and address the same.

**Comment by Abhey** [comment]
@stoty One question for you, I created a Docker Image with all the Pre-Requisites for testing the Phoenix-Spark connector. Do you think it's a good idea to add the reference to the same in the official documentation?

Repository Link -
* https://github.com/Abhey/phoenix-spark-connector-demo
